github-code-reviews/code-reviews.md

```diff
@@ -0,0 +1,102 @@
+- **Knowledge Transfer** - When a team makes code reviews part of the development process, they ensure knowledge of each feature is shared among multiple members of the team. 
+  - This minimizes disruption when a member transfers, finds new employment, retires, etc. 
+  - It also makes it easier for other team members to adapt or adjust the feature under review in the future.
+  - Code reviews provide a useful tool for newer team members to explore and learn about the codebase. That said, a new team member should **not** be the only reviewer.
+  - Finally, code reviews allow developers to learn techniques, libraries, and tricks from other members of the team.
+- **Architecture** - By involving several team members in a proposed change or feature, a code review allows the team to examine the architecture of a feature and judge it in the larger context of the application. Will the change scale?  Is it flexible and extensible? Does it match or integrate with the architecture of the larger application?
+- **Finding Bugs** - As we might expect, having more eyes on code early in development allows the team to catch more bugs **before** they enter production, lowering costs and speeding up development time. 
+- **Verify Against Requirements** - The team can check the submission against the requirements and verify that the code satisfies them.
+
+## Types Of Code Reviews
+
+There are several different ways to review code.
+
+### Pull Requests
+
+Github and other version management software systems provide methods to review suggested changes before accepting them into the `main` branch. This is called a _Pull Request_ or PR. A PR allows team members to comment on and propose changes on a branch before it is merged in. 
+
+### Pair Programming
+
+Another way to provide a review on code is through _pair programming_. In pair programming, two developers work on the feature as a team. The developers take turns with one _driving_, or writing the code and focusing on the immediate task at hand. Meanwhile, the second developer _navigates_ or looks at the work from a higher level guiding around complications and obstacles the driver cannot focus on in the moment.
+
+### Mob Programming
+
+A third method is called [_mob programming_](https://en.wikipedia.org/wiki/Mob_programming) where a team of developers works on the same feature, at the same time, together. One developer drives while the rest of the team navigates. Mob programming can be very useful on complicated, tricky, or core features where wide understanding on the team is essential. Sometimes it is useful to have as many eyes on the code as possible.
+
+Both mob programming and pair programming can also be done remotely using software such as [Visual Studio's Liveshare extension](https://code.visualstudio.com/blogs/2017/11/15/live-share). They also have the advantage of a very short feedback loop, while a more common pull request review happens asynchronously and can happen hours or even days later. 
```

Comment:
I wonder about moving this section to the end since the sections after this that talk about Code Reviews are more specifically talking about PRs